server/homebrew.api.js

```diff
@@ -89,148 +136,151 @@ const newBrew = async (req, res)=>{
+		});
 	if(!saved) return;
 
-	if(!res.headersSent) return res.status(200).send(saved);
+	res.status(200).send(saved);
+};
+
+const deleteGoogleBrew = async (account, id, editId, res)=>{
+	const auth = await GoogleActions.authCheck(account, res);
+	await GoogleActions.deleteGoogleBrew(auth, id, editId);
+	return true;
 };
 
 const deleteBrew = async (req, res)=>{
-	if(req.params.id.length > 12) {
-		const deleted = await deleteGoogleBrew(req.account, req.params.id, res)
+	const { brew, account } = req;
+	if(brew.googleId) {
+		const googleDeleted = await deleteGoogleBrew(account, brew.googleId, brew.editId, res)
 			.catch((err)=>{
+				console.error(err);
 				res.status(500).send(err);
 			});
-		if(deleted) return res.status(200).send();
-	} else {
-		const deleted = await deleteLocalBrew(req.account, req.params.id)
-			.catch((err)=>{
-				res.status(err.status).send(err.message);
-			});
-		if(deleted) return res.status(200).send(deleted);
-		return res.status(200).send();
+		if(!googleDeleted) return;
 	}
-};
 
-const deleteLocalBrew = async (account, id)=>{
-	const brew = await HomebrewModel.findOne({ editId: id });
-	if(!brew) {
-		throw { status: 404, message: 'Can not find homebrew with that id' };
-	}
-
-	if(account) {
-		// Remove current user as author
-		brew.authors = _.pull(brew.authors, account.username);
-		brew.markModified('authors');
-	}
+	if(brew._id) {
+		if(account) {
+			// Remove current user as author
+			brew.authors = _.pull(brew.authors, account.username);
+			brew.markModified('authors');
+		}
 
-	if(brew.authors.length === 0) {
-		// Delete brew if there are no authors left
-		await brew.remove()
-			.catch((err)=>{
-				console.error(err);
-				throw { status: 500, message: 'Error while removing' };
-			});
-	} else {
-		// Otherwise, save the brew with updated author list
-		return await brew.save()
-			.catch((err)=>{
-				throw { status: 500, message: err };
-			});
+		if(brew.authors.length === 0) {
+			// Delete brew if there are no authors left
+			await brew.remove()
+				.catch((err)=>{
+					console.error(err);
+					throw { status: 500, message: 'Error while removing' };
+				});
+		} else {
+			// Otherwise, save the brew with updated author list
+			await brew.save()
+				.catch((err)=>{
+					throw { status: 500, message: err };
+				});
+		}
 	}
-};
 
-const deleteGoogleBrew = async (account, id, res)=>{
-	const auth = await GoogleActions.authCheck(account, res);
-	await GoogleActions.deleteGoogleBrew(auth, id);
-	return true;
+	res.status(204).send();
 };
 
 router.post('/api', asyncHandler(newBrew));
-router.put('/api/:id', asyncHandler(updateBrew));
-router.put('/api/update/:id', asyncHandler(updateBrew));
-router.delete('/api/:id', asyncHandler(deleteBrew));
-router.get('/api/remove/:id', asyncHandler(deleteBrew));
-
-module.exports = router;
+router.put('/api/:id', asyncHandler(getBrew('edit')), asyncHandler(updateBrew));
```

Comment:
`getBrew` performs the fetch from both the database and Google and combining those objects. Since at the time we don't know if the user is transferring to/from google or if the doc is google or not I think it's pretty important there. I can look into removing the outer `if(!brew)` piece though

Reply:
Well, for `update`, we hit the outer `if(!brew)` anyway and don't retrieve anything, don't we? Since we are already sending in the updated brew from the client?

Reply:
So for `update`, the incoming brew is actually on `req.body`, whereas `getBrew` is referring to `req.brew`, which is set by either this middleware, or earlier middlewares. I'm pretty sure I had put that outer if in because there's a couple spots where it hits it because an earlier middleware had already set `req.brew`. I'm wondering if it may have been a dev artifact though, and I might be able to remove it. I'll play around with it and let you know

Reply:
Ok, so let me step through this:

1) User sends `req.brew` = updated brew
2) We `getBrew()`, so `req.body` = old brew
3) We immediately overwrite the old brew with the updated brew with `let brew = _.assign(req.brew, excludePropsFromUpdate(req.body));` The content of the old brew is lost at this point, but we now have a Mongo object.
4) We retrieve the old brew a second time with `HomebrewModel.findOne()` and immediately overwrite it again, and we still have a Mongo object.
5) We save the brew.

Seems like we can skip steps 2 and 3. Especially since `getBrew` does all the extra steps of looking for Google brews and stuff.

Reply:
The `if (!brew)` is in fact a development artifact, removing it!